forage-android/src/main/java/com/joinforage/forage/android/core/services/forageapi/network/ForageApiResponse.kt

```diff
@@ -97,101 +95,17 @@ sealed class ForageApiResponse<out T> {
      * handle the error and display the appropriate
      * [customer-facing message](https://docs.joinforage.app/docs/document-error-messages#template-error-table-for-fns-documentation).
      */
-    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
-        internal companion object {
-            fun fromError(error: ForageError): Failure {
-                return Failure(listOf(error))
-            }
-        }
-    }
-}
-
-/**
- * A model that represents an error response from the Forage API.
- *
- * @property httpStatusCode A number that corresponds to the HTTP status code that the Forage API
- * returns in response to the request.
- * @property code A short string that helps identify the cause of the error.
- * For example, [`ebt_error_55`](https://docs.joinforage.app/reference/errors#ebt_error_55)
- * indicates that a customer entered an invalid EBT Card PIN.
- * @property message A string that specifies developer-facing error handling instructions.
- * @property details An object that includes additional details about the error, when available, like for
- * [`ebt_error_51`](https://docs.joinforage.app/reference/errors#ebt_error_51) (Insufficient Funds).
- * @see [SDK Errors](https://docs.joinforage.app/reference/errors#sdk-errors) for a comprehensive
- * list of error `code` and `message` pairs.
- *
- */
-data class ForageError(
-    val httpStatusCode: Int,
-    val code: String,
-    val message: String,
-    val details: ForageErrorDetails? = null
-) {
-    /**
-     * A method that converts the [ForageError] response to a string.
-     *
-     * @return A string representation of a [ForageError] instance, including code, message,
-     * HTTP status, and any other available details from the Forage API.
-     */
-    override fun toString(): String {
-        return "Code: $code\nMessage: $message\nStatus Code: $httpStatusCode\nError Details (below):\n$details"
-    }
-}
-
-internal data class ForageErrorObj(
-    val code: String,
-    val message: String
-) {
-    object ForageErrorObjMapper {
-        fun from(string: String): ForageErrorObj {
-            val jsonObject = JSONObject(string)
+    data class Failure internal constructor(val error: ForageError) : ForageApiResponse<Nothing>() {
 
-            val code = jsonObject.getString("code")
-            val message = jsonObject.getString("message")
+        // support legacy list of single error
+        val errors: List<ForageError> = listOf(error)
```

Comment:
can we make this louder by using `@Deprecated tag`, like we [did here](https://github.com/teamforage/forage-android-sdk/blob/d0ae5d2baa45ce32e07e74e958cadf79eb32c322/forage-android/src/main/java/com/joinforage/forage/android/core/ui/element/ForagePinElement.kt#L138)

Reply:
Yup! That sounds good to me!

Reply:
just did this! going to mark as resolved